<launch>
	<!-- Example finding 3D poses of the objects detected -->

	<!-- $roslaunch openni_launch openni.launch depth_registration:=true -->
	<include file="$(find openni2_launch)/launch/openni2.launch">
		<arg name = "depth_registration" value = "true"/>
	</include>		

	<rosparam command="load" file="$(find object_recognition)/config/object_orientation.yaml" param="object_config"/>

	<node name="find_object_3d" pkg="find_object_2d" type="find_object_2d" output="screen">
		<param name="gui" value="false" type="bool"/>
		<param name="settings_path" value="$(find object_recognition)/settings/object_recognition.ini" type="str"/>
		<!--<param name="settings_path" value="~/.ros/find_object_2d.ini" type="str"/>-->
		<param name="subscribe_depth" value="true" type="bool"/>
		<param name="objects_path" value="$(find object_recognition)/objects/" type="str"/>
		<param name="object_prefix" value="object" type="str"/>

		<remap from="rgb/image_rect_color" to="camera/rgb/image_rect_color"/>
		<remap from="depth_registered/image_raw" to="camera/depth_registered/image_raw"/>
		<remap from="depth_registered/camera_info" to="camera/depth_registered/camera_info"/>
	</node>
	
	<node name="object_listener" pkg="object_recognition" type="object_recognition" output="screen"/>

	
</launch>
